lib/index.js

```diff
@@ -87,3 +92,5 @@ module.exports = ({ redis, keyPrefix, ...defaults }) => {
   };
 
 };
+
+module.exports.GCRA_LUA = GCRA_LUA;
```

Comment:
one thought - instead of exposing the lua script like this, could you instead make a function like:

```
module.exports.defineNodeRedisScripts = (nodeRedis) => {
  return {
    performGcraRateLimit: nodeRedis.defineScript({
      NUMBER_OF_KEYS: 1,
      SCRIPT: GCRA_LUA,
      transformArguments(key, now, burst, rate, period, cost) {
        return [key, now.toString(), burst.toString(), rate.toString(), period.toString(), cost.toString()];
      },
      transformReply(reply) {
        return reply;
      }
    })
  };
};
```

which could then be used by someone like this:

```
const redis = Redis.createClient({
  scripts: {
    ...RedisGCRA.defineNodeRedisScripts(Redis),
    /* other custom user scripts defined here */
  }
});
```

That way if we ever have to change details about how the script works (or for some reason even add an additional script), users don't have to worry about updating their implementation.

Reply:
Certainly could, however there's a reason I didn't: Typescript. We actually use typescript ourselves, so would like to preserve the ability to have a strongly typed function available to us. So for our use case, the script would look something like this:

```
module.exports.defineNodeRedisScripts = (nodeRedis) => {
  return {
    performGcraRateLimit: nodeRedis.defineScript({
      NUMBER_OF_KEYS: 1,
      SCRIPT: GCRA_LUA,
      transformArguments(key: string, now: number, burst: number, rate: number, period: number, cost: number) {
        return [key, now.toString(), burst.toString(), rate.toString(), period.toString(), cost.toString()];
      },
      transformReply(reply) {
        return [reply[0] as boolean, reply[1] as number, reply[2] as number, reply[3] as number, reply[4] as string, reply[5] as string;
      }
    })
  };
};
```

I did think of biting off a typescript conversion too, but am a bit short on time for this. If there is a desire, I'd be happy to include this suggestion as part of that process!

Reply:
how about best of both worlds - *both* expose GCRA_LUA and add the function. That way you can continue to build the performGcraRateLimit manually if you want, but still gives the option of cleaning up usage for other folks.